replica/compaction_group.hh

```diff
@@ -57,6 +57,8 @@ class compaction_group {
     seastar::condition_variable _staging_done_condition;
     // Gates async operations confined to a single group.
     seastar::named_gate _async_gate;
+    // Gates flushes.
+    seastar::named_gate _flush_gate;
```

Comment:
nit: In the future, `named_gate` should be initialized with a name so we can know which kind of gate it was in case of exception.